pkg/stream/pipeline.go

```diff
@@ -149,11 +151,17 @@ func (p *Pipeline) process(ctx context.Context, force bool) {
 	}()
 
 	p.ticker.Stop()
-	messages, err := p.callback.Process(ctx, p.batch)
 
-	if err != nil {
-		p.logger.Error(err, "could not process the batch")
-		return
+	var err error
+	var messages = p.batch
+
+	for _, stage := range p.stages {
+		messages, err = stage.Process(ctx, messages)
+
+		if err != nil {
+			p.logger.Error(err, "could not process the batch")
```

Comment:
should we note in which state it failed as well? Like a name for the stage or so?

Reply:
Maybe we should add that later on. For now, I would assume, that the error specifies enough information.